src/main/java/ome/services/RawPixelsBean.java

```diff
@@ -682,9 +687,18 @@ public synchronized Map<Integer, int[]> getHistogram(int[] channels,
             int binCount, boolean globalRange, PlaneDef plane) {
         errorIfNotLoaded();
 
-        if (requiresPixelsPyramid())
-            throw new ApiUsageException(
-                    "This method can not handle tiled images yet.");
+        //Find resolution level closest to max plane size without
+        //exceeding it
```

Comment:
What happens in the case of a single-resolution image that exceeds the maximum plane size e.g. a large image without pyramidal levels imported in a server where PixelData is turned off? Should this throw an exception rather than trying to use the zero-th level resolution?

Reply:
That's probably a good idea.

Reply:
@sbesson I'm actually having trouble getting to this scenario. If I import a large fake file with PixelData-0 turned off and then try to get a histogram for it, I get a `omero.MissingPyramidException` from `RawPixelsBean::setPixelsId` (it tries to instantiate a `PixelBuffer`) before the histogram code even gets called. Are there scenarios where a single-resolution image exceeds max plane size but we don't get a `MissingPyramidException` from the `PixelsService`?